hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -210,16 +210,19 @@ public RpcClient(ConfigurationSource conf, String omServiceId)
       ozoneManagerProtocolClientSideTranslatorPB.setS3AuthCheck(isS3);
       if (isS3) {
         // S3 Auth works differently and needs OM version to be at 2.0.0
-        String omVersion = conf.get(OZONE_OM_CLIENT_PROTOCOL_VERSION_KEY);
-        if (!validateOmVersion(omVersion, serviceInfoEx.getServiceInfoList())) {
+        OzoneManagerVersion minOmVersion = conf.getEnum(
```

Comment:
I think this should be run for all clients, not just S3 gateway ones, since the OM version is given to all clients requesting service info. S3g would specify the non-default version, so the effect would be the same.